File: backend/MedicalHistory.Application/Services/PatientService.cs
using MedicalHistory.Core.Interfaces.Repositories;
using MedicalHistory.Core.Interfaces.Services;
using MedicalHistory.Core.Models;

namespace MedicalHistory.Application.Services;

public class PatientService : IPatientService
{
    public readonly IPatientRepository _patientRepository;

    public PatientService(IPatientRepository patientRepository)
    {
        _patientRepository = patientRepository;
    }

    public async Task<Guid> CreatePatient(Patient patient)
    {
        return await _patientRepository.Create(patient);
    }

    public async Task<Guid> UpdatePatient(Patient patient)
    {
        return await _patientRepository.Update(patient);
    }

    public async Task<Guid> DeletePatient(Guid id)
    {
        return await _patientRepository.Delete(id);
    }

    public async Task<Patient> GetPatient(Guid id)
    {
        var patient = await _patientRepository.Get(id);
        return patient;
    }

    public async Task<List<Patient>> GetAllPatients()
    {
        return await _patientRepository.GetAll();
    }
}

File: frontend/src/services/patientService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/patient';

class PatientService {
  async getAllPatients() {
    const response = await axios.get(API_BASE_URL);
    return response.data;
  }

  async getPatientById(id) {
    const response = await axios.get(`${API_BASE_URL}/${id}`);
    return response.data;
  }

  async createPatient(patientData) {
    const response = await axios.post(API_BASE_URL, patientData);
    return response.data;
  }

  async updatePatient(id, patientData) {
    const response = await axios.put(`${API_BASE_URL}/${id}`, patientData);
    return response.data;
  }

  async deletePatient(id) {
    const response = await axios.delete(`${API_BASE_URL}/${id}`);
    return response.data;
  }
}

export default new PatientService();